radis/levels/partfunc.py

```diff
@@ -1136,6 +1137,93 @@ def _at(self, T):
         return np.interp(T, self.T_range, self.Q_range)
 
 
+class PartFuncKurucz(RovibParFuncTabulator):
+    """Return partition function using interpolation of tabulated values of local file kuruczpartfn.txt
+
+    Parameters
+    ----------
+    atom: str
+        exomol isotope full name
+    ionization_state : int
+
+    """
+
+    def __init__(self, atom, ionization_state):
+        super(PartFuncKurucz, self).__init__()
+        if ionization_state == "00":
+            self.key = atom + "_I"
+        elif ionization_state == "01":
+            self.key = atom + "_II"
+        else:
+            self.key = atom + "_III"
+        # Load data in constructor
+        path_partfn = join(getProjectRoot(), "db", "kuruczpartfn.txt")
+        pfdat = pd.read_csv(path_partfn, sep="\s+", header=None)
+        self.pfdat = pfdat.set_index(0)
+        # Locate the row for the specific atom and ionization state
+        pf_atom = self.pfdat.loc[f"{self.key}"]
+        self.pfT_values = np.array(
+            [
+                1e-05,
+                1e-04,
+                1e-03,
+                1e-02,
+                1e-01,
+                1.5e-01,
+                2e-01,
+                3e-01,
+                5e-01,
+                7e-01,
+                1.0,
+                1.3,
+                1.7,
+                2.0,
+                3.0,
+                5.0,
+                7.0,
+                10.0,
+                15.0,
+                20.0,
+                30.0,
+                50.0,
+                70.0,
+                100.0,
+                130.0,
+                170.0,
+                200.0,
+                250.0,
+                300.0,
+                500.0,
+                700.0,
+                1e03,
+                1.5e03,
+                2e03,
+                3e03,
+                4e03,
+                5e03,
+                6e03,
+                7e03,
+                8e03,
+                9e03,
+                1e04,
+            ]
+        )
+        self.pf_values = pf_atom.values[0:].astype(
+            float
+        )  # Exclude the first value (it's the atomic number)
+
+    def _at(self, T):
+        # Interpolate to find the partition function at the desired temperature
+        try:
+            return np.interp(T, self.pfT_values, self.pf_values)
```

Comment:
> I added the range test

Nice, see comment on commit.

>  I am not sure to understand what you mean by including the Kurucz _at(self,T) function to the existing one 

There is already a function call `_at(self, T)` in this file. It would make a lot of sense to me that the function `_at(self, T)` is always called when a partition function is evaluated. Then:

- species == molecule and the code runs as usual 
- species == atom and it's your portion that is executed.

Reply:
The thing is that for now by mentioning the parfuncfmt in SpectrumFactory directly, the user selects earlier which Partfunc Class will be used and thus which _at function will be used. So what you suggested is to gather all the _at functions  or at least the Kurucz and Exomol ones (they are very similar actually only the temperature range checks are different) under one same method which can be applied to both atoms and molecules ? @minouHub